tests/testUtils.py

```diff
@@ -245,18 +247,23 @@ def writeCSV(self, data, fileName):
 
     {'id1' : {'f1':n1, 'f2':n2}, 'id2' : {'f1':n3, 'f2':n4}}
     """
-    csvFile = open(fileName, 'w')
-    csvFileWriter = csv.writer(csvFile)
-    # get the headers from the first row
-    header = sorted(data[list(data.keys())[0]].keys())
-    header = ['testCase'] + header
-    csvFileWriter.writerow(header)
-    for testCase in sorted(data.keys()):
-      thisCase = data[testCase]
-      thisCase['testCase'] = testCase
-      row = []
-      for h in header:
-        row = row + [thisCase.get(h, "N/A")]
-      csvFileWriter.writerow(row)
-    csvFile.close()
-    self._logger.info('Wrote to file %s', fileName)
+    # Get the headers from the first testCase in _testedSet
+    # If no tests were run, the length of _testedSet will be 0, and no files should be written
+    if len(self._testedSet) > 0:
+      csvFile = open(fileName, 'w')
+      csvFileWriter = csv.writer(csvFile, lineterminator='\n')
+      testedCases = sorted(self._testedSet)
+      header = sorted(data[testedCases[0]].keys())
+      header = ['testCase'] + header
+      csvFileWriter.writerow(header)
+      for testCase in testedCases:
+        thisCase = data[testCase]
+        thisCase['testCase'] = testCase
+        row = []
+        for h in header:
+          row = row + [thisCase.get(h, "N/A")]
+        csvFileWriter.writerow(row)
+      csvFile.close()
+      self._logger.info('Wrote to file %s', fileName)
```

Comment:
Not necessarily. As it is a logging message, a format is applied by default. So both would work.